sdk/src/test/java/com/finbourne/lusid/integration/HoldingsTests.java

```diff
@@ -0,0 +1,271 @@
+        requests.add(testDataUtiliites.buildTransactionRequest(this.instrumentIds.get(2), 100.0, 103.0, currency, day1, "Buy"));
+
+        //  add transactions on T+5, increasing the amount of instrument 1
+        requests.add(testDataUtiliites.buildTransactionRequest(this.instrumentIds.get(1), 100.0, 104.0, currency, day2, "Buy"));
+        requests.add(testDataUtiliites.buildTransactionRequest(this.instrumentIds.get(3), 100.0, 105.0, currency, day2, "Buy"));
+
+        //  add the trades
+        this.transactionPortfoliosApi.upsertTransactions(SCOPE, portfolioCode, requests);
+
+        //  get the holds on T+10
+        VersionedResourceListOfPortfolioHolding holdings = this.transactionPortfoliosApi.getHoldings(SCOPE, portfolioCode, false, day3,
+                null, null, null, null, null, null);
+
+        holdings.getValues().sort(Comparator.comparing(PortfolioHolding::getInstrumentUid));
+
+        assertThat(holdings.getCount(), is(equalTo(5)));
+
+        //  cash balance
+        assertThat(holdings.getValues().get(0).getInstrumentUid(), is(equalTo("CCY_" + currency)));
+        assertThat(holdings.getValues().get(0).getHoldingType(), is(equalTo("B")));    //  B = balance
+        assertThat(holdings.getValues().get(0).getUnits(), is(equalTo(48500.0)));
+
+        //  instrument holdings, holding type 'P' represents a position
+        assertThat(holdings.getValues().get(1).getInstrumentUid(), is(equalTo(this.instrumentIds.get(0))));
+        assertThat(holdings.getValues().get(1).getHoldingType(), is(equalTo("P")));
+        assertThat(holdings.getValues().get(1).getUnits(), is(equalTo(100.0)));
+        assertThat(holdings.getValues().get(1).getCost().getAmount(), is(equalTo(10100.0)));
+
+        assertThat(holdings.getValues().get(2).getInstrumentUid(), is(equalTo(this.instrumentIds.get(1))));
+        assertThat(holdings.getValues().get(2).getHoldingType(), is(equalTo("P")));
+        assertThat(holdings.getValues().get(2).getUnits(), is(equalTo(200.0)));   //  2 transactions
+        assertThat(holdings.getValues().get(2).getCost().getAmount(), is(equalTo(20600.0)));
+
+        assertThat(holdings.getValues().get(3).getInstrumentUid(), is(equalTo(this.instrumentIds.get(2))));
+        assertThat(holdings.getValues().get(3).getHoldingType(), is(equalTo("P")));
+        assertThat(holdings.getValues().get(3).getUnits(), is(equalTo(100.0)));
+        assertThat(holdings.getValues().get(3).getCost().getAmount(), is(equalTo(10300.0)));
+
+        assertThat(holdings.getValues().get(4).getInstrumentUid(), is(equalTo(this.instrumentIds.get(3))));
+        assertThat(holdings.getValues().get(4).getHoldingType(), is(equalTo("P")));
+        assertThat(holdings.getValues().get(4).getUnits(), is(equalTo(100.0)));
+        assertThat(holdings.getValues().get(4).getCost().getAmount(), is(equalTo(10500.0)));
+
+    }
+
+    @Test
+    public void set_target_holdings() throws ApiException {
```

Comment:
This would be more applicable if it set the holdings first (ie take-on) then had subsequent transactions